src/node_shapes.jl

```diff
@@ -0,0 +1,227 @@
+"""
+    node_shape(shape::Symbol, clip::Bool; <keyword arguments>)
+
+Create an outline for the graph node. Also, can define a clipping region for the node region.
+
+Returns a dictionary of drawing options and a callable draw function (in this order).
+
+# Arguments
+- `shape::Symbol`: Specify a node shape among `:circle`, `:rectangle` or a `:polygon` (to be implemented).
+- `clip::Bool`: Define a clipping region over the region defined by `shape`.
+
+# Keywords
+- `radius`: Specify if shape is `:circle`.
+- `width`: Specify if shape is `:rectangle`.
+- `height`: Specify if shape is `:rectangle`.
+"""
+function node_shape(shape::Symbol, clip::Bool=false; kwargs...)
+    kw = Dict(collect(kwargs)...)
+    opts = Dict{Symbol, Any}()
+    opts[:shape] = shape
+    opts[:clip] = clip
+    if shape == :rectangle
+        opts[:width] = get(kw, :width, 5)
+        opts[:height] = get(kw, :height, 5)
+    elseif shape == :circle
+        opts[:radius] = get(kw, :radius, 5)
+    end
+    draw = (video, object, frame; shape=:circle, position=O, clip=false, kwargs...) -> begin
+        opts = Dict(collect(kwargs)...)
+        if shape == :rectangle
+            w, h = get(opts, :width, 5), get(opts, :height, 5)
+            rect(position[1]-w/2, position[2]-h/2, w, h, clip ? :clip : :stroke)
+            bounding_box = (position-(w/2, h/2), position+(w/2, h/2))
+        elseif shape == :circle
+            r = get(opts, :radius, 5)
+            circle(position, r, clip ? :clip : :stroke)
+            bounding_box = (position-r, position+r)
+        end
+        object.meta.opts[:bounding_box] = bounding_box
+    end
+    return opts, draw
+end
+
+"""
+    node_text(text::AbstractString, align::Symbol, color::String, offset::Real)
+    node_text(text::LaTeXString, align::Symbol, color::String, offset::Real)
+
+Label a node with text or latex.
+
+# Arguments
+- `align::Symbol`: Specify the alignment of the text label with respect to the node.
+    - This position is determined with respect to the node `bounding_box` returned as a drawing option by [`node_shape`](@ref).
+    - Can take the values `:inside`, `:top`, `:bottom`, `:left`, `:right`. Default is `:inside`.
+- `color`: The color of the text. Default is "black".
+- `offset`: Used only when `align` is not `:inside`. Specifies an offset in the direction of `align`.
+"""
+function node_text(text::AbstractString, align::Symbol, color::String="black", offset::Real=1)
```

Comment:
I think we want to have an option to specify the font size?

Reply:
To specify font and size, it is possible to add a (args...; kwargs...) -> setfont(<font>, <size>) anywhere in the list before a `node_text` is created. Since these functions are supposed to be backup options if the user wants to skip some drawing properties I skipped that option. Do you think it is still needed?

Reply:
I think font size is something so standard that we want to include it in `node_text`

Reply:
Okay sure will do that.